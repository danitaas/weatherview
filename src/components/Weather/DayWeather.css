

.DayWeather-container {
  display: flex; /* todo: flex grid if target browser supports it */
  flex-direction: row;
  width: auto;
  flex-wrap: wrap;
  padding: 10px;
}

.DayWeather-item {
  width: 180px; /* todo: flex grid if target browser supports it */
  margin-bottom: 10px;
}