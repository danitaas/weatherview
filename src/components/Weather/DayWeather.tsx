import * as React from 'react';
import './DayWeather.css';

// todo: naming conventions/data formats
export interface IProps {
    day?: string;
    mintemp?: string;
    maxtemp?: string;
    conditions?: string;
    wind?: string;
}

function DayWeather(props: IProps) {
    if (props.day == "error") {
        throw new Error('An error occured');
    }
    return (
        <div className="DayWeather-container">
            <div className="DayWeather-item">
                <div>{props.day}</div>
                <div>{props.mintemp}</div>
                <div>{props.maxtemp}</div>
                <div>{props.conditions}</div>
                <div>{props.wind}</div>
            </div>
        </div>
    );
}

export default DayWeather;
