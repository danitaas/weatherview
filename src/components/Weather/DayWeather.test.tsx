import * as React from 'react';
import * as ReactDOM from 'react-dom';
import DayWeather from './DayWeather';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DayWeather />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders all props', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DayWeather day={"a"} mintemp={"a"} maxtemp={"a"} conditions={"a"} wind={"a"} />, div);
    ReactDOM.unmountComponentAtNode(div);
});