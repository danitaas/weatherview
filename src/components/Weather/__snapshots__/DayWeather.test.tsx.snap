// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div>
  Invalid data
</div>
`;

exports[`renders snapshot with props 1`] = `
<div
  className="DayWeather-container"
>
  <div
    className="DayWeather-item"
  >
    Wednesday 27
  </div>
  <div
    className="DayWeather-item"
  >
    26째C/78째F
  </div>
  <div
    className="DayWeather-item"
  >
    25째C/78째F
  </div>
  <div
    className="DayWeather-item"
  >
    a
  </div>
  <div
    className="DayWeather-item"
  >
    4kmh/2mph N
  </div>
</div>
`;
